heft-plugins/heft-serverless-stack-plugin/src/ServerlessStackPlugin.ts

```diff
@@ -5,142 +5,137 @@ import * as path from 'path';
-  IHeftPlugin,
-  IHeftStringParameter,
-  ScopedLogger
+  IHeftTaskSession,
+  IHeftTaskPlugin,
+  IHeftTaskRunHookOptions,
+  IScopedLogger
 } from '@rushstack/heft';
-// TODO: Expose this API
-import { SubprocessTerminator } from '@rushstack/heft/lib/utilities/subprocess/SubprocessTerminator';
-import { FileSystem, Import } from '@rushstack/node-core-library';
+import { FileSystem, Import, SubprocessTerminator } from '@rushstack/node-core-library';
+import type {
+  PluginName as Webpack4PluginName,
+  IWebpackPluginAccessor as IWebpack4PluginAccessor
+} from '@rushstack/heft-webpack4-plugin';
+import type {
+  PluginName as Webpack5PluginName,
+  IWebpackPluginAccessor as IWebpack5PluginAccessor
+} from '@rushstack/heft-webpack5-plugin';
 
-const PLUGIN_NAME: string = 'ServerlessStackPlugin';
-const TASK_NAME: string = 'heft-serverless-stack';
+const PLUGIN_NAME: 'ServerlessStackPlugin' = 'ServerlessStackPlugin';
+const WEBPACK_PLUGIN_NAME: typeof Webpack4PluginName & typeof Webpack5PluginName = 'WebpackPlugin';
 const SST_CLI_PACKAGE_NAME: string = '@serverless-stack/cli';
 
-/**
- * Options for `ServerlessStackPlugin`.
- *
- * @public
- */
-export interface IServerlessStackPluginOptions {}
-
-/** @public */
-export class ServerlessStackPlugin implements IHeftPlugin<IServerlessStackPluginOptions> {
-  public readonly pluginName: string = PLUGIN_NAME;
-
-  private _logger!: ScopedLogger;
-
-  public apply(
-    heftSession: HeftSession,
-    heftConfiguration: HeftConfiguration,
-    options: IServerlessStackPluginOptions
-  ): void {
-    this._logger = heftSession.requestScopedLogger(TASK_NAME);
-
-    const sstParameter: IHeftFlagParameter = heftSession.commandLine.registerFlagParameter({
-      associatedActionNames: ['test', 'build', 'start'],
-      parameterLongName: '--sst',
-      // Once https://github.com/serverless-stack/serverless-stack/issues/1537 is fixed, we may be
-      // eliminate the need for this parameter.
-      description: 'Invokes the SST postprocessing - requires AWS credentials'
-    });
+export default class ServerlessStackPlugin implements IHeftTaskPlugin {
+  private _logger!: IScopedLogger;
+
+  public apply(taskSession: IHeftTaskSession, heftConfiguration: HeftConfiguration): void {
+    this._logger = taskSession.logger;
+
+    // Once https://github.com/serverless-stack/serverless-stack/issues/1537 is fixed, we may be
+    // eliminate the need for this parameter.
+    const sstParameter: CommandLineFlagParameter = taskSession.getFlagParameter('--sst');
+    const sstStageParameter: CommandLineStringParameter = taskSession.getStringParameter('--sst-stage');
+
+    // Only tap if the --sst flag is set.
+    if (sstParameter.value) {
+      const configureWebpackTapOptions: { name: string; stage: number } = {
+        name: PLUGIN_NAME,
+        stage: Number.MAX_SAFE_INTEGER
+      };
+      const configureWebpackTap: () => Promise<null> = async () => {
+        this._logger.terminal.writeVerboseLine(
```

Comment:
```suggestion
        this._logger.terminal.writeLine(
```

Seems like this should always be logged.